docs/grammar.txt

```diff
@@ -24,10 +24,12 @@ StructDefinition = 'struct' Identifier '{'
 ModifierDefinition = 'modifier' Identifier ParameterList?  ( 'virtual' | OverrideSpecifier )* Block
 ModifierInvocation = Identifier ( '(' ExpressionList? ')' )?
 
-FunctionDefinition = 'function' Identifier? ParameterList
+FunctionDefinition = FunctionDescriptor ParameterList
                      ( ModifierInvocation | StateMutability | 'external' | 'public' | 'internal' | 'private' | 'virtual' | OverrideSpecifier )*
                      ( 'returns' ParameterList )? ( ';' | Block )
 
+FunctionDescriptor = 'function' Identifier? | 'constructor' | 'fallback' | 'receive'
```

Comment:
```suggestion
FunctionDescriptor = 'function' Identifier | 'constructor' | 'fallback' | 'receive'
```
Now that the `fallback` function needs to be explicitly defined via `fallback`, shouldn't it be like this /\?

Reply:
You're totally right, @leonardoalt!
Completely let it ran past me because the parser Diligence is maintaining has as a goal to be able to parse any solidity version without needing versioning itself. So we're basically just supporting all old syntaxes for simplicity's sake.

I'll change this right away.